[arshinsikka] iP

arshinsikka opened the pull request:
# 🚀 ShinBot: A Smart Task Manager  

> _"Your mind is for having ideas, not holding them."_ – David Allen ([source](https://gettingthingsdone.com))

ShinBot is a **smart chatbot** designed to help users manage tasks efficiently. It is:

- 📝 **Text-based**
- ⚡ **Lightweight & Fast**
- 🎯 **Easy to Use**

## 🔹 Features  
- ✅ Task management  
- ⏳ Deadlines *(coming soon)*  
- 🔔 Reminders *(coming soon)*  

## 📌 How to Use
1. **Download** the latest release from [here](https://github.com/arshinsikka/ip/releases).
2. **Run it** using the command:
   ```sh
   java -jar shin.jar
3. Add tasks & deadlines using simple commands.
4. Manage your tasks effortlessly! 🎉

💻 Sample Code
public class Main {
    public static void main(String[] args) {
        System.out.println("Welcome to ShinBot!");
    }
}

✅ To-Do List
 Implement Task Management
 Add Deadline Support
 Implement Notifications (Upcoming Feature!)
 GUI Support (Planned Update)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

arshinsikka committed:
Level-0: Added greeting and exit functionality

arshinsikka committed:
Level-1: Added echo functionality

arshinsikka committed:
Level-2: Added task storage and listing functionality

arshinsikka committed:
Level-3: Added ability to mark tasks as done and undone using Task class

arshinsikka committed:
Level-4: Added support for ToDos, Deadlines, and Events using inheritance

arshinsikka committed:
A-TextUiTesting: Added automated UI testing with I/O redirection

arshinsikka committed:
Level-5: Added error handling for empty descriptions and unknown commands using custom exceptions

arshinsikka committed:
Final: Implemented full task management with error handling and automated testing

arshinsikka committed:
A-Enums: Implemented TaskType enum for better task categorization

arshinsikka committed:
Final version before PR

arshinsikka committed:
Level-7: Implemented File I/O for Save and Load

arshinsikka committed:
Merge Level-7 into master

Ernestnzx commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,72 @@
+import java.io.*;
```
Wildcard imports are not allowed, explicitly declare the classes that you want to use.

Ernestnzx commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    protected String by;
```
Small nit, perhaps a more descriptive string name?

Ernestnzx commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    protected String from;
+    protected String to;
```
Small nit, perhaps a more descriptive name for the start and end timings for the `Event` class?

Ernestnzx reviewed (commented):
Overall, I found the code style in this repository easy to read and 
follows the convention imposed [at](https://se-education.org/guides/conventions/java/intermediate.html#naming). 

Though I would like some of the variables could be a little more
descriptive for ease of review. Good job and keep it up!

arshinsikka committed:
Finalize Level 8: Dates and Times

arshinsikka committed:
Refactor code to follow OOP principles (A-MoreOOP)

arshinsikka committed:
Refactored code into packages and fixed IntelliJ run issues

arshinsikka committed:
Merge branch 'add-gradle-support'

arshinsikka committed:
Gradle setup complete and chatbot runs successfully

arshinsikka committed:
Added JUnit tests for TaskList and Deadline

arshinsikka committed:
Added JavaDoc comments

arshinsikka committed:
Applied Java coding standard fixes

blacksesamezongzi commented on src/main/java/shin/Storage.java:
```diff
@@ -0,0 +1,101 @@
+            String description = taskDetails[2].trim();
+            Task task = null;
+
+            switch (taskType) {
+                case "T":
```
Should the case be un-indented?

blacksesamezongzi commented on src/main/java/shin/Ui.java:
```diff
@@ -0,0 +1,64 @@
+        this.scanner = new Scanner(System.in);
+    }
+
+    public void showWelcome() {
+        System.out.println("____________________________________________________________");
+        System.out.println("Hello! I'm Shin");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________________________");
+    }
```
Should descriptions added before this method?

blacksesamezongzi commented on src/main/java/shin/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+/**
+ * Represents a task with a deadline.
+ */
+
```
Perhaps you need to leave no line break between the description and the method section?

blacksesamezongzi commented on src/main/java/shin/task/Event.java:
```diff
@@ -0,0 +1,30 @@
+        this.to = LocalDate.parse(to);
+    }
+
+    public LocalDate getFrom() {   // ✅ Add this getter
+        return from;
+    }
```
I like the way you explain the reason you make this change in your commit.

blacksesamezongzi reviewed (commented):
LGTM. Just a few nits to fix.

arshinsikka committed:
Add Level-10: GUI implementation

arshinsikka committed:
Merge branch 'branch-Level-10'

arshinsikka committed:
A-Assertions: Add assertions to validate critical assumptions in code

arshinsikka committed:
A-Streams: Refactor code to use Java Streams API

arshinsikka committed:
Merge branch 'branch-A-Streams'

arshinsikka committed:
Shin: Add randomized chatbot responses

arshinsikka committed:
Error handling: Improve file loading and input validation

arshinsikka committed:
Testing: Increase test coverage for Deadline, TaskList, and Todo

arshinsikka committed:
Tag project with A-AiAssisted

arshinsikka committed:
Deadline: Refactor date formatting with AI-assisted utility method

arshinsikka committed:
Docs: Add README with AI usage acknowledgment

arshinsikka committed:
Code Style: Replace wildcard imports with explicit imports

arshinsikka committed:
Fix: Ensure send button and Enter key trigger input correctly

arshinsikka committed:
Enhancement: Improve chatbot responses with more contextual handling

arshinsikka committed:
docs: Add Ui.png and update README for GitHub Pages

arshinsikka committed:
Update README with project details

arshinsikka committed:
Update README with project details

arshinsikka committed:
Update README with project details

arshinsikka committed:
Update README with project details

arshinsikka committed:
Refactor methods to improve SLAP and reduce method length

Several methods in Shin.java and related classes exceed reasonable lengths and violate SLAP. Long methods decrease readability and make the code harder to maintain.

To improve code quality, let's:
* Extract smaller helper methods from long methods such as getResponse() and run() in Shin.java.
* Reduce deep nesting in conditionals for better readability.
* Ensure methods adhere to Single Level of Abstraction (SLAP) principles.

These changes make the code more modular and easier to understand without altering functionality.

arshinsikka committed:
Refactor methods to improve SLAP and reduce method length

Several methods in Shin.java and related classes exceed reasonable lengths and violate SLAP. Long methods decrease readability and make the code harder to maintain.

To improve code quality, let's:
* Extract smaller helper methods from long methods such as getResponse() and run() in Shin.java.
* Reduce deep nesting in conditionals for better readability.
* Ensure methods adhere to Single Level of Abstraction (SLAP) principles.

These changes make the code more modular and easier to understand without altering functionality.

arshinsikka committed:
Docs: Add User Guide and UI screenshot

Add a structured User Guide in docs/README.md.
This guide includes:
* Detailed explanations for all commands.
* Installation steps and how to run Shin.
* Frequently Asked Questions (FAQ) section.
* A preview image (Ui.png) showcasing the chatbot interface.

GitHub Pages is enabled for hosting the User Guide, ensuring users can easily access it.

arshinsikka committed:
Update README.md for better structure and clarity

The previous README.md had an unstructured Features section, long descriptions, and lacked readability in the usage commands.

To improve user experience, let's:
* Reformat the Features section for better clarity.
* Use tables to display usage commands in an easy-to-read manner.
* Shorten explanations while retaining necessary details.
* Add a structured FAQ section.
* Ensure proper Markdown formatting for consistency.

These changes make the README more structured, professional, and user-friendly while improving readability for new users.

arshinsikka committed:
Refactor SLAP usage in task handling methods

Shortened long methods and improved readability by extracting helper methods. Enhances maintainability and clarity without altering functionality.

arshinsikka committed:
Improve SLAP compliance in command processing

Refactored getResponse() and run() methods to follow SLAP principles. Extracted helper methods to reduce deep nesting and improve readability.

arshinsikka committed:
Fix coding standard violations

Fix various Java coding standard issues across the project.

- Enforce proper naming conventions (camelCase, PascalCase, SCREAMING_SNAKE_CASE).
- Use K&R style brackets and add spaces around operators.
- Replace wildcard imports with explicit imports.
- Add missing Javadoc comments for public methods.

These changes improve code readability and maintainability.

arshinsikka committed:
Update usage commands in README for better formatting

The table-based approach caused formatting issues on GitHub Pages.
Replaced it with a text-based format using headings and examples.
This ensures the commands are displayed clearly and consistently for users.
